src/mscorlib/shared/System/Threading/Tasks/ValueTask.cs

```diff
@@ -79,51 +80,15 @@ public ValueTask(IValueTaskSource source, short token)
             _obj = source;
             _token = token;
 
-            _flags = 0;
+            _continueOnCapturedContext = true;
         }
 
-        /// <summary>Non-verified initialization of the struct to the specified values.</summary>
-        /// <param name="obj">The object.</param>
-        /// <param name="token">The token.</param>
-        /// <param name="flags">The flags.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private ValueTask(object obj, short token, ValueTaskFlags flags)
+        private ValueTask(object obj, short token, bool continueOnCapturedContext)
         {
             _obj = obj;
             _token = token;
-            _flags = flags;
-        }
-
-        /// <summary>Gets whether the contination should be scheduled to the current context.</summary>
-        internal bool ContinueOnCapturedContext
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get => (_flags & ValueTaskFlags.AvoidCapturedContext) == 0;
-        }
-
-        /// <summary>Gets whether the object in the <see cref="_obj"/> field is a <see cref="Task"/>.</summary>
-        internal bool ObjectIsTask
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get => (_flags & ValueTaskFlags.ObjectIsTask) != 0;
-        }
-
-        /// <summary>Returns the <see cref="Task"/> stored in <see cref="_obj"/>.  This uses <see cref="Unsafe"/>.</summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal Task UnsafeGetTask()
-        {
-            Debug.Assert(ObjectIsTask);
-            Debug.Assert(_obj is Task);
-            return Unsafe.As<Task>(_obj);
-        }
-
-        /// <summary>Returns the <see cref="IValueTaskSource"/> stored in <see cref="_obj"/>.  This uses <see cref="Unsafe"/>.</summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal IValueTaskSource UnsafeGetValueTaskSource()
-        {
-            Debug.Assert(!ObjectIsTask);
-            Debug.Assert(_obj is IValueTaskSource);
-            return Unsafe.As<IValueTaskSource>(_obj);
+            _continueOnCapturedContext = true;
```

Comment:
>true; [](start = 41, length = 5)

this should be = continueOnCapturedContext?

Reply:
Ugh. Yes.  That suggests we also have a test hole then, as these passed.  Will take a look.